Build failure: bluesky-pds

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

`nix build "github:NixOS/nixpkgs/nixos-unstable#bluesky-pds"`

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/build/source/service/node_modules/.pnpm/better-sqlite3@10.1.0/node_modules/better-sqlite3/build'
  TOUCH ba23eeee118cd63e16015df367567cb043fed872.intermediate
  ACTION deps_sqlite3_gyp_locate_sqlite3_target_copy_builtin_sqlite3 ba23eeee118cd63e16015df367567cb043fed872.intermediate
  TOUCH Release/obj.target/deps/locate_sqlite3.stamp
  CC(target) Release/obj.target/sqlite3/gen/sqlite3/sqlite3.o
rm -f Release/obj.target/deps/sqlite3.a Release/obj.target/deps/sqlite3.a.ar-file-list; mkdir -p `dirname Release/obj.target/deps/sqlite3.a`
ar crs Release/obj.target/deps/sqlite3.a @Release/obj.target/deps/sqlite3.a.ar-file-list
  COPY Release/sqlite3.a
  CXX(target) Release/obj.target/better_sqlite3/src/better_sqlite3.o
In file included from /nix/store/pq5996xinzcns0k5la4xhmlwzwgz2w6s-nodejs-24.12.0-source/deps/v8/include/cppgc/common.h:8,
                 from /nix/store/pq5996xinzcns0k5la4xhmlwzwgz2w6s-nodejs-24.12.0-source/deps/v8/include/v8.h:23,
                 from /nix/store/pq5996xinzcns0k5la4xhmlwzwgz2w6s-nodejs-24.12.0-source/src/node.h:74,
                 from ./src/better_sqlite3.lzz:11,
                 from ../src/better_sqlite3.cpp:4:
/nix/store/pq5996xinzcns0k5la4xhmlwzwgz2w6s-nodejs-24.12.0-source/deps/v8/include/v8config.h:13:2: error: #error "C++20 or later required."
   13 | #error "C++20 or later required."
      |  ^~~~~
In file included from /nix/store/pq5996xinzcns0k5la4xhmlwzwgz2w6s-nodejs-24.12.0-source/deps/v8/include/v8-array-buffer.h:12,
                 from /nix/store/pq5996xinzcns0k5la4xhmlwzwgz2w6s-nodejs-24.12.0-source/deps/v8/include/v8.h:24:
/nix/store/pq5996xinzcns0k5la4xhmlwzwgz2w6s-nodejs-24.12.0-source/deps/v8/include/v8-local-handle.h:280:5: warning: identifier 'requires' is a keyword in C++20 [-Wc++20-compat]
  280 |     requires std::is_base_of_v<T, S>
      |     ^~~~~~~~
```

### Additional context

Overriding `nodejs` back to `nodejs_22` fixes build failure.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.2, NixOS, 26.05 (Yarara), 26.05.20251230.cad22e7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+1`
 - channels(root): `"nixos-25.05"`
 - nixpkgs: `/nix/store/lv4kcr2y4b9k4ynhkga157507zxf372b-source`

### Notify maintainers

@t4ccer @isabelroses

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/bl/bluesky-pds/package.nix
+++ b/pkgs/by-name/bl/bluesky-pds/package.nix
@@ -8,7 +8,7 @@
   fetchPnpmDeps,
   pnpmConfigHook,
   fetchFromGitHub,
-  nodejs,
+  nodejs_22,
   vips,
   pkg-config,
   nixosTests,
@@ -18,6 +18,9 @@
 }:
 
 let
+  # build failure against better-sqlite3, so we use nodejs_22; upstream
+  # bluesky-pds uses 20
+  nodejs = nodejs_22;
   nodeSources = srcOnly nodejs;
   pythonEnv = python3.withPackages (p: [ p.setuptools ]);
 in

